spotify_import.py

```diff
@@ -73,46 +75,59 @@ def _save_tracks_to_library(self, tracks):
             self.sp.current_user_saved_tracks_add(tracks)
             print(f'Added {len(tracks)} tracks to library')
 
-    def _save_tracks(self, tracks: List[str], failed_count: int):
+    def _save_tracks(self, tracks: List[str], failed_count: int, playlist: Optional[dict] = None):
         if self.destination == 'playlist':
-            playlist = self.sp.user_playlist_create(user=self._get_user_id(), name=self.playlist, public=False)
+            if playlist is None:
+                playlist = self.sp.user_playlist_create(user=self._get_user_id(), name=self.playlist, public=False)
             self.sp.playlist_add_items(playlist['id'], tracks)
         else:
             self._save_tracks_to_library(tracks)
-        print(f'Saved a total of {len(tracks)} tracks to {self.destination}, '
+        print(f'[work in progress..] Saved a total of {len(tracks)} tracks to {self.destination}, '
               f'failed to add {failed_count} songs (see failed.txt)')
 
     def _run_txt(self):
-        with open(self.songs) as songs_file, open('failed.txt', 'w') as failed_file:
+        playlist = self.sp.user_playlist_create(user=self._get_user_id(), name=self.playlist, public=False)
```

Comment:
`self.playlist` does not exist if the `destination` is 'library', so the script fails when run with destination=library. Need to add a check for destination and pass in a default value (e.g. `None`) instead of `self.playlist` if the destination is library.